tinywl/tinywl.zig

```diff
@@ -207,7 +207,13 @@ const Server = struct {
             const scene_surface = wlr.SceneSurface.fromBuffer(scene_buffer) orelse return null;
 
             var it: ?*wlr.SceneNode = node;
-            while (it) |n| : (it = n.parent) {
+            while (it) |n| : (it = blk: {
+                if (n.parent) |parent| {
+                    break :blk &parent.node;
+                } else {
+                    break :blk null;
+                }
+            }) {
```

Comment:
It would be cleaner to make `it` have type `?*wlr.SceneTree` and start iteration at `node.parent`.